front/lib/actions/mcp_internal_actions/wrappers.ts

```diff
@@ -74,7 +74,7 @@ export function withToolLogging<T>(
       };
     }
 
-    logger.info(loggerArgs, "Tool execution start");
+    logger.info.bind(logger)(loggerArgs, "Tool execution start");
```

Comment:
I don't really understand what that does, but I'll trust you that it does something good :)

So the whole idea is to get more complete logging at this point, and not to fix the issue. So we'll have to go back to the user and ask him to try it again so we can learn from his new failure. Is that the idea?